File: FC.App/Scripts/Shared/Models/UCountry.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var FC;
(function (FC) {
    var Shared;
    (function (Shared) {
        var Models;
        (function (Models) {
            var UCountry = (function (_super) {
                __extends(UCountry, _super);
                function UCountry(c) {
                    _super.call(this, c);
                    if (c) {
                        this.Name = c.Name;
                        this.CultureIsoName = c.CultureIsoName;
                        this.LanguageName = c.LanguageName;
                        this.RegionInfo = c.RegionInfo;
                    }
                }
                return UCountry;
            }(Models.BaseModel));
            Models.UCountry = UCountry;
        })(Models = Shared.Models || (Shared.Models = {}));
    })(Shared = FC.Shared || (FC.Shared = {}));
})(FC || (FC = {}));

File: FC.Shared/Entities/UCountry.cs
using FC.Interfaces.Data;
using FC.Shared.EntityMapper;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FC.Shared.Entities
{
    
    public class UCountry : BaseModel
    {
        [Key]
        public Guid? CountryID { get; set; }

        public bool IsPopular { get; set; }
       
        public UCountry()
        {
        }

        public UCountry(object country):
            base(country) 
        {
            UCountry c = (UCountry)country;
            this.Name = c.Name;
            this.CultureIsoName = c.CultureIsoName;
            this.LanguageName = c.LanguageName;
        }
        public string CultureIsoName { get; set; }
        public string LanguageName { get; set; }
        public string Currency { get; set; }
        
    }
}
